File: files2proc.rb
class AgeOfFiles
require 'fileutils'
require 'net/smtp'

Dir.chdir("/path/to/Directory")  
#point attention at the "filestoprocess" directory (above)

Dir.foreach(".") do |f|                                     #step through each file=f       
  next if f == '.' or f == '..'                             #unwanted directories
  time_now = Time.now
  original_time = File.mtime(f)                             #modified times of each file
  age_of_file = ((time_now - original_time) / 60).to_i      #age in minutes, to_i=no decimal
    if age_of_file >= 5
      then FileUtils.copy(f, "/home/talbot/backup"); 
        FileUtils.move(f, "/home/talbot/Desktop");     #drop into production
          puts "Dropped #{f} back into production and backed it up on /home/talbot/backup/" 
      else puts "All clear: #{f} is still young at #{age_of_file} minute(s) old!"
    end
  end
msg = <<EOF
Subject: URGENT--Files stranded
Message
EOF

smtp = Net::SMTP.new("mail.domain.net", 25)
smtp.start		#previous two lines opens a connection to mail server
smtp.sendmail msg, 'talbot@mail.com', 'talbot@mail.com'
end

File: files2proc.pl
#!/usr/bin/perl
    use strict;
    use warnings;
    use File::stat;
    use Time::localtime;
    use Mail::Mailer;
#watch the directory for existing files in intervals of 10 minutes = process, service, scheduled task

my $processDir = "/path/to/watch/";
my $productionDir = "/path/to/dump/";
my @oldFiles = ();

chdir $processDir or die "Can't find process: $!\n";
my @files = <*>;

    opendir (DIR, $processDir) || die "Nice try Dingus: $!\n";
	foreach my $file(@files) {
 	my $modDate = ctime(stat($file)->mtime);
	my $minuteDiff = int((time - stat($file)->mtime) / 60);		#the file's age in minutes
	if ($minuteDiff >= 5) {
	    push(@oldFiles, $file);
	    print "$file in $processDir is $minuteDiff minutes old\n";
	} 
     }

my $joinFiles = join(", ", @oldFiles);

if (@oldFiles) {
my $message = "Type a message here.";

my $mailer = Mail::Mailer->new("smtp", Server => 'mailserver.domain.net');
$mailer->open( { From => 'Talbot Lawrence <talbotlawrence@mail.com>',
		To => 'Talbot Lawrence <talbotlawrence@mail.com>',
		Subject => 'URGENT' } ) || die "mailer->open failed: $!\n";

print $mailer $message; 
$mailer->close || die "can't close mailer: $!\n";
} 

    closedir(DIR);
    exit 0;
